### PR TITLE
Enhance theme with `toggleGroup.button.kind`

### LINKED ISSUE #7407: ToggleGroup — Add ability to define button kind or border on the button
### Is this something you or your team is interested in contributing a pull request towards?
Yes.

### Expected Behavior

Should be able to define a button "kind" for individual ToggleGroup buttons, or define button object styling.

### Actual Behavior

Unable to do this.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

The current styling of ToggleGroup buttons has some built in logic for how the individual buttons should behave within their container. This PR gives the caller more control of the individual button styling by allowing them to:
- Define a `kind` via string, as a reference to one of their pre-defined button kinds

For ease of reviewing, I've staged some themes in preview.js to demo the string reference and object behaviors (changes to SpaceX story and preview.js should be undone before merging).

Using the [SpaceX story](https://deploy-preview-7409--sad-tereshkova-173d07.netlify.app/?path=/story/data-data-spacex--space-x&globals=theme:hpe), use the theme picker to switch into the various theme approaches (see [preview.js](https://github.com/grommet/grommet/pull/7409/files#diff-dcc18ca391a4d099c9c938796dd2b6553c7513f15c22e7147173a9365244d2a4) for the code that defines this).

#### Where should the reviewer start?
src/js/components/ToggleGroup/ToggleGroup.js

#### What testing has been done on this PR?
Locally, see SpaceX story and jest tests added.

#### How should this be manually tested?
SpaceX story.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes #7407 

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by MikeKingdom — APPROVED** [review]
Looks good

**Review by britt6612 — APPROVED** [review]
Looks good, 

agree on waiting though on the desired design direction then maybe make the changes more lean with just the `string` or `object` I think if there is not a need for both changes then we go with the one.

**Comment by taysea** [comment]
> Looks good,
> 
> agree on waiting though on the desired design direction then maybe make the changes more lean with just the `string` or `object` I think if there is not a need for both changes then we go with the one.

Connected with Joelle, the styling will be aligned between togglegroup buttons and individual toolbar buttons, so only keeping support for `string` for now.

File: src/js/themes/base.d.ts [hunk 1]
```diff
@@ -183,6 +183,7 @@ interface ButtonType {
     };
   };
   gap?: GapType;
+  background?: BackgroundType;
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/themes/base.d.ts:186]: Updated.

File: src/js/themes/base.d.ts [hunk 2]
```diff
@@ -1813,6 +1814,7 @@ export interface ThemeType {
   };
   toggleGroup?: {
     button?: {
+      kind?: string;
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/themes/base.d.ts:1925]: Updated.

**Review by halocline — APPROVED** [review]

**Pull request merged**